osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -81,9 +81,10 @@ protected virtual Track GetVirtualTrack(double emptyLength = 0)
         /// <returns>The applicable <see cref="IBeatmapConverter"/>.</returns>
         protected virtual IBeatmapConverter CreateBeatmapConverter(IBeatmap beatmap, Ruleset ruleset) => ruleset.CreateBeatmapConverter(beatmap);
 
-        public virtual IBeatmap GetPlayableBeatmap(RulesetInfo ruleset, IReadOnlyList<Mod> mods = null, TimeSpan? timeout = null)
+        public virtual IBeatmap GetPlayableBeatmap(RulesetInfo ruleset, IReadOnlyList<Mod> mods = null, TimeSpan? timeout = null, CancellationToken timeoutToken = default)
```

Comment:
I'd also call this `cancellationToken` or just `token` rather than `timeoutToken`.